File: frontend/start.js
#!/usr/bin/env node

/**
 * Startup script for the Roof Design Validation Frontend
 */

const fs = require('fs');
const path = require('path');
const { spawn, execSync } = require('child_process');

function checkEnvironment() {
    console.log('Checking environment configuration...');
    
    // Check if .env.local exists
    const envFile = path.join(__dirname, '.env.local');
    if (!fs.existsSync(envFile)) {
        console.log('.env.local file not found. Creating from template...');
        
        const envTemplate = `NEXT_PUBLIC_BACKEND_URL=http://localhost:8000`;
        
        fs.writeFileSync(envFile, envTemplate);
        console.log('Created .env.local file');
    }
    
    console.log('Environment configuration ready');
    return true;
}

function checkDependencies() {
    console.log('Checking dependencies...');
    
    // Check if node_modules exists
    if (!fs.existsSync(path.join(__dirname, 'node_modules'))) {
        console.log('node_modules not found. Installing dependencies...');
        
        try {
            console.log('Running npm install...');
            execSync('npm install', { stdio: 'inherit', cwd: __dirname });
            console.log('Dependencies installed successfully');
            return true;
        } catch (error) {
            console.error('Failed to install dependencies:', error.message);
            return false;
        }
    }
    
    console.log('Dependencies are ready');
    return true;
}

function startServer() {
    console.log('Starting Roof Design Validation Frontend...');
    console.log('Frontend will be available at: http://localhost:3000');
    console.log('Press Ctrl+C to stop the server');
    console.log('-'.repeat(50));
    
    const dev = spawn('npm', ['run', 'dev'], {
        stdio: 'inherit',
        cwd: __dirname,
        shell: true
    });
    
    dev.on('close', (code) => {
        if (code === 0) {
            console.log('Frontend server stopped cleanly');
        } else {
            console.log(`Frontend server exited with code ${code}`);
        }
    });
    
    dev.on('error', (error) => {
        console.error('Failed to start frontend server:', error.message);
    });
    
    // Handle graceful shutdown
    process.on('SIGINT', () => {
        console.log('\n Shutting down frontend server...');
        dev.kill('SIGINT');
        process.exit(0);
    });
}

function main() {
    console.log('Roof Design Validation Frontend');
    console.log('='.repeat(40));
    
    // Check environment
    if (!checkEnvironment()) {
        process.exit(1);
    }
    
    // Check and install dependencies
    if (!checkDependencies()) {
        console.log('Dependency setup failed. Please install manually:');
        console.log('   npm install');
        process.exit(1);
    }
    
    // Start development server
    startServer();
}

if (require.main === module) {
    main();
}

File: backend/start.py
#!/usr/bin/env python3
"""
Startup script for the Roof Design Validation Backend
"""

import os
import sys
import subprocess
from pathlib import Path
from dotenv import load_dotenv

def check_environment():
    """Check if environment is properly configured"""
    
    print("🔍 Checking environment configuration...")
    
    # Check if .env file exists
    env_file = Path('.env')
    if not env_file.exists():
        print(".env file not found. Creating from template...")
        
        # Create .env from template
        env_template = """# OpenAI Configuration
OPENAI_API_KEY=your_openai_api_key_here
OPENAI_API_BASE=https://api.openai.com/v1
OPENAI_MAIN_MODEL=gpt-5
OPENAI_SUMMARY_MODEL=gpt-5-mini

# Server Configuration  
HOST=127.0.0.1
PORT=8000
DEBUG=true
ENVIRONMENT=development

# CORS Configuration
ALLOWED_ORIGINS=http://localhost:3000,http://127.0.0.1:3000

# Cost Tracking
TRACK_API_COSTS=true
LOG_LEVEL=INFO"""
        
        with open('.env', 'w') as f:
            f.write(env_template)
        
        print("Created .env file from template")
        print("Please edit .env file and add your OpenAI API key")
        return False
    
    # Check if API key is set
    load_dotenv()
    
    api_key = os.getenv('OPENAI_API_KEY')
    if not api_key or api_key == 'your_openai_api_key_here':
        print("OPENAI_API_KEY not set in .env file")
        print("Please edit the .env file and add your OpenAI API key")
        return False
    
    print("Environment configuration looks good")
    return True

def install_dependencies():
    """Install Python dependencies"""
    
    print("Installing dependencies...")
    
    try:
        subprocess.run([sys.executable, '-m', 'pip', 'install', '-r', 'requirements.txt'], 
                      check=True, capture_output=True)
        print("Dependencies installed successfully")
        return True
    except subprocess.CalledProcessError as e:
        print(f"Failed to install dependencies: {e}")
        return False

def start_server():
    """Start the FastAPI server"""
    
    print("Starting Roof Design Validation API server...")
    print("API will be available at: http://localhost:8000")
    print("API documentation: http://localhost:8000/docs")
    print("Press Ctrl+C to stop the server")
    print("-" * 50)
    
    try:
        # Import and run server
        import uvicorn
        from config import settings
        
        uvicorn.run(
            "main:app",
            host=settings.host,
            port=settings.port,
            reload=settings.debug,
            log_level=settings.log_level.lower()
        )
    except KeyboardInterrupt:
        print("\n Server stopped by user")
    except Exception as e:
        print(f"Server failed to start: {e}")

def main():
    """Main startup function"""
    
    print("Roof Design Validation Backend")
    print("=" * 40)
    
    # Change to backend directory if needed
    backend_dir = Path(__file__).parent
    os.chdir(backend_dir)
    
    # Check environment
    if not check_environment():
        sys.exit(1)
    
    # Skip automatic dependency installation for conda environments
    print("Skipping automatic dependency installation...")
    print("Make sure you have installed dependencies manually with conda/pip")
    
    # Start server
    start_server()

if __name__ == "__main__":
    main()